File: DevExpress-Demo/CS/EditorsDemo.Wpf/Modules/FilterBehavior/FilterBehaviorListBoxEditViewModel.cs
using DevExpress.DemoData.Models;
using DevExpress.Mvvm;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
#if DXCORE3
using Microsoft.EntityFrameworkCore;
#else
using System.Data.Entity;
#endif

namespace EditorsDemo.FilterBehavior {
    public class FilterBehaviorListBoxEditViewModel : ViewModelBase {
        public IList<Product> Products { get; private set; }
        public List<string> Categories { get; private set; }

        public FilterBehaviorListBoxEditViewModel() {
            var context = new NWindContext();
            Products = context.Products
                .Include(x => x.Category)
                .ToList();
            Categories = context.Categories
                .Select(category => category.CategoryName)
                .ToList();
        }
    }
}

File: DevExpress-Demo/VB/EditorsDemo.Wpf/Modules/FilterBehavior/FilterBehaviorListBoxEditViewModel.vb
Imports DevExpress.DemoData.Models
Imports DevExpress.Mvvm
Imports System.Collections.Generic
Imports System.Linq
#If DXCORE3
using Microsoft.EntityFrameworkCore;
#Else
Imports System.Data.Entity

#End If
Namespace EditorsDemo.FilterBehavior

    Public Class FilterBehaviorListBoxEditViewModel
        Inherits ViewModelBase

        Private _Products As IList(Of DevExpress.DemoData.Models.Product), _Categories As List(Of String)

        Public Property Products As IList(Of Product)
            Get
                Return _Products
            End Get

            Private Set(ByVal value As IList(Of Product))
                _Products = value
            End Set
        End Property

        Public Property Categories As List(Of String)
            Get
                Return _Categories
            End Get

            Private Set(ByVal value As List(Of String))
                _Categories = value
            End Set
        End Property

        Public Sub New()
            Dim context = New NWindContext()
            Products = context.Products.Include(Function(x) x.Category).ToList()
            Categories = context.Categories.[Select](Function(category) category.CategoryName).ToList()
        End Sub
    End Class
End Namespace
